crates/test-utils/src/chain/deploy.rs

```diff
@@ -0,0 +1,1050 @@
+                    U256::from(1),
+                    [0x02; 32],
+                )
+                .send()
+                .await?;
+            factory
+                .deploy_target(
+                    [0x02; 32],
+                    steth_deployer_coordinator.address(),
+                    pool_config.clone(),
+                    Vec::new().into(),
+                    config.steth_hyperdrive_fixed_apr,
+                    config.steth_hyperdrive_time_stretch_apr,
+                    U256::from(2),
+                    [0x02; 32],
+                )
+                .send()
+                .await?;
+            factory
+                .deploy_target(
+                    [0x02; 32],
+                    steth_deployer_coordinator.address(),
+                    pool_config.clone(),
+                    Vec::new().into(),
+                    config.steth_hyperdrive_fixed_apr,
+                    config.steth_hyperdrive_time_stretch_apr,
+                    U256::from(3),
+                    [0x02; 32],
+                )
+                .send()
+                .await?;
+            factory
+                .deploy_target(
+                    [0x02; 32],
+                    steth_deployer_coordinator.address(),
+                    pool_config.clone(),
+                    Vec::new().into(),
+                    config.steth_hyperdrive_fixed_apr,
+                    config.steth_hyperdrive_time_stretch_apr,
+                    U256::from(4),
+                    [0x02; 32],
+                )
+                .send()
+                .await?;
+            let tx = factory
+                .deploy_and_initialize(
+                    [0x02; 32],
+                    steth_deployer_coordinator.address(),
+                    pool_config,
+                    Vec::new().into(),
+                    config.steth_hyperdrive_contribution,
+                    config.steth_hyperdrive_fixed_apr,
+                    config.steth_hyperdrive_time_stretch_apr,
+                    Options {
+                        as_base: true,
+                        destination: address,
+                        extra_data: Vec::new().into(),
+                    },
+                    [0x02; 32],
+                )
+                .value(config.steth_hyperdrive_contribution)
+                .send()
+                .await?
+                .await?
+                .unwrap();
+            let logs = tx
+                .logs
+                .into_iter()
+                .filter_map(|log| {
+                    if let Ok(HyperdriveFactoryEvents::DeployedFilter(log)) =
+                        HyperdriveFactoryEvents::decode_log(&log.into())
+                    {
+                        Some(log)
+                    } else {
+                        None
+                    }
+                })
+                .collect::<Vec<_>>();
+            logs[0].clone().hyperdrive
+        };
```

Comment:
What do you think about abstracting out the deployment/initialization of hyperdrive instances to a separate function? The processes seem relatively similar and mechanical for the current instances... would also be a great place to handle adding to the registry. 

Cool with punting on this if we expect dramatic changes just worth thinking about